docs/rfc/0004_component_groups_scale_view.md

```diff
@@ -0,0 +1,300 @@
+a `RisingWaveScaleView` that targets the frontend groups.
+
+> What is the expected outcome?
+
+1. A CR `RisingWaveScaleView` and its controller, with the following features:
+    - Users can target any of the four components in it, but aren't able to change it after.
+    - Users can target one or more valid groups in it, but aren't able to change it after.
+    - Users can set the priorities of groups for scaling out and in separately.
+    - It reflects all the fields required by the scale subresource described above and keeps them synced with the
+      RisingWave's.
+    - It locks the target RisingWave object on the specified groups and prevents the direct modifications (either update
+      or delete) on the related fields.
+        - It acquires the lock at the creation and removes it before deletion.
+
+2. An improvement on `RisingWave` resource and validating webhook to support lock of groups:
+    - Lock fields in the status, which is structured and easy to lookup.
+    - Enhanced validating webhook to reject any changes from besides the controller of `RisingWaveScaleView`.
+
+# **Explanation**
+
+> This is the technical portion of the RFC. Explain the design in sufficient detail.
+
+## `RisingWaveScaleView` CRD
+
+```yaml
+apiVersion: risingwave.risingwavelabs.com/v1alpha1
+kind: RisingWaveScaleView
+metadata:
+  name: hello-frontend-scale-view
+spec:
+  targetRef:
+    name: hello             # Name of the RisingWave object, in the same namespace.
+    component: frontend     # One of meta, frontend, compute, and compactor.
+  replicas: 10              # Target replicas. It will be synced from the RisingWave object when the controller grabs the lock.
+  # Strict verification mode on replicas:
+  #   1. If enabled, the validating webhook will reject any invalid changes on .spec.replicas (e.g., exceeding the min/max range)
+  #   2. If disabled, the mutating webhook will adjust the .spec.replicas to be enclosed by the range.
+  # Serialized label selector. Would be set by the webhook.
+  strict: false
+  labelSelector: "risingwave/name=hello,risingwave/component=frontend,risingwave/group in (enduring,spot)"
+  scalePolicy: # An array of groups and the policies for scale, 
+  # optional and empty means the default group with the default policy.
+  # The order here indicates the priorities.
+
+  - group: 'enduring'       # Target group name
+    # 0-10, optional. The groups will be sorted by the priority and the current replicas. 
+    # The higher it is, the more replicas of the target group will be considered kept, i.e. scale out first, scale in last.
+    priority: 1
+    replicas: 2             # Controlled field, maintained by the mutating webhook. Current desired replicas.
+    constraints:
+      min: 1              # >= 0, optional, default is 0
+      max: 2              # >= minimum, optional, default is 1000 (indicates infinite because 1000 Pods are many enough to be considered as infinite).
+  - group: 'spot'
```

Comment:
Does this example mean it will first try to follow `enduring` group and try `spot` group if it fails?

Reply:
I guess it's demonstrated in the following section 🤣